src/ctapipe/monitoring/calculator.py

```diff
@@ -0,0 +1,327 @@
+        config: traitlets.loader.Config
+            Configuration specified by config file or cmdline arguments.
+            Used to set traitlet values.
+            This is mutually exclusive with passing a ``parent``.
+        parent: ctapipe.core.Component or ctapipe.core.Tool
+            Parent of this component in the configuration hierarchy,
+            this is mutually exclusive with passing ``config``
+        stats_aggregator: ctapipe.monitoring.aggregator.StatisticsAggregator
+            The ``StatisticsAggregator`` to use. If None, the default via the
+            configuration system will be constructed.
+        """
+        super().__init__(subarray=subarray, config=config, parent=parent, **kwargs)
+        self.subarray = subarray
+
+        # Initialize the instances of StatisticsAggregator
+        self.stats_aggregator = {}
+        if stats_aggregator is None:
+            for _, _, name in self.stats_aggregator_type:
+                self.stats_aggregator[name] = StatisticsAggregator.from_name(
+                    name, subarray=self.subarray, parent=self
+                )
+        else:
+            name = stats_aggregator.__class__.__name__
+            self.stats_aggregator_type = [("type", "*", name)]
+            self.stats_aggregator[name] = stats_aggregator
+
+        # Initialize the instances of OutlierDetector
+        self.outlier_detectors = {}
+        if self.outlier_detector_list is not None:
+            for outlier_detector in self.outlier_detector_list:
+                self.outlier_detectors[
+                    outlier_detector["apply_to"]
+                ] = OutlierDetector.from_name(
+                    name=outlier_detector["name"],
+                    validity_range=outlier_detector["validity_range"],
+                    subarray=self.subarray,
+                    parent=self,
+                )
+
+    def first_pass(
+        self,
+        table,
+        tel_id,
+        masked_pixels_of_sample=None,
+        col_name="image",
+    ) -> Table:
+        """
+        Calculate the monitoring data for a given set of events with non-overlapping aggregation chunks.
+
+        This method performs the first pass over the provided data table to calculate
+        various statistics for calibration purposes. The statistics are aggregated with
+        non-overlapping chunks (``chunk_shift`` set to None), and faulty pixels are detected
+        using a list of outlier detectors.
+
+
+        Parameters
+        ----------
+        table : astropy.table.Table
+            DL1-like table with images of shape (n_images, n_channels, n_pix), event IDs and
+            timestamps of shape (n_images, )
+        tel_id : int
+            Telescope ID for which the calibration is being performed
+        masked_pixels_of_sample : ndarray, optional
+            Boolean array of masked pixels of shape (n_pix, ) that are not available for processing
```

Comment:
`n_pixels`? Or `(n_channels, n_pixels)`? In principle, high gain and low gain can be independently fail.

Reply:
Also, please use `n_pixels`

Reply:
This boolean mask corresponds to hardware failure, so i think it should be (n_pixels,). We treat high and low gain independently later on software level when aggregating the stats for each type of calibration.

Reply:
But the hardware of the two channels is at least partly independent, at least in LST (same PMT but e.g. different DRS4 chips). So they can also fail independently.

Reply:
Actually the shape is `(n_channels, n_pixels)` as you say. It's propagated to the [aggregator](https://github.com/cta-observatory/ctapipe/blob/main/src/ctapipe/monitoring/aggregator.py#L167C50-L167C73) (`np.ma.array(images, mask=masked_pixels_of_sample)`), where `masked_pixels_of_sample` should have the same shape as `images` (which is of shape `(n_channels, n_pixels)`). I will update all the docstrings accordingly.

With my current setup using [`lstchain-data_r0_to_dl1` script](https://github.com/cta-observatory/cta-lstchain/blob/main/lstchain/scripts/lstchain_data_r0_to_dl1.py) (for writing R1 files for calib events) and `ctapipe-process` (for integrating and extracting DL1a images for calib events), the information about broken pixels due to hardware failure is lost. Once we write in ACADA DL0 files (cosmic and calib events) we should have access to this information.